--- a/server/metin2/Source/Server/game/src/packet.h
+++ b/server/metin2/Source/Server/game/src/packet.h
@@ -702,7 +702,8 @@
 	HEADER_GC_KEY_AGREEMENT_COMPLETED = 250, // 250
 	HEADER_GC_KEY_AGREEMENT = 251, // 251
 #endif
-	HEADER_GC_TIME_SYNC = 252, // 252
+	HEADER_GC_HANDSHAKE_OK = 252, // 252 (0xfc)
+	HEADER_GC_TIME_SYNC = 252, // legacy alias (handshake ok)
 	HEADER_GC_PHASE = 253, // 253
 	HEADER_GC_BINDUDP = 254, // 254
 	HEADER_GC_HANDSHAKE = 255, // 255

