--- Server/game/src/char_item.cpp
+++ Server/game/src/char_item.cpp
@@ -11379,6 +11379,12 @@
 	if (MediumItem == NULL || BaseItem == NULL || MaterialItem == NULL)
 		return;
 
+	if (BaseItem->GetCount() > 1 || MaterialItem->GetCount() > 1)
+	{
+		ChatPacket(CHAT_TYPE_INFO, LC_STRING("Split the items first. You cannot use stacked items here."));
+		return;
+	}
+
 	if (MediumItem->GetType() != ITEM_MEDIUM)
 		return;
 
