--- a/server/metin2/Source/Server/game/src/input.cpp
+++ b/server/metin2/Source/Server/game/src/input.cpp
@@ -81,6 +81,8 @@
 			iPacketLen = 1;

 		else if (!m_pPacketInfo->Get(bHeader, &iPacketLen, &c_pszName))

 		{

+			c_pszName = "UNKNOWN";

+			r_iBytesProceed = m_iBufferLeft; // flush remaining bytes to avoid log-spam loops

 			sys_err("UNKNOWN HEADER: %d, LAST HEADER: %d(len:%d), %s, REMAIN BYTES: %d, fd: %d",

 				bHeader, bLastHeader, iLastPacketLen, c_pszName, m_iBufferLeft, lpDesc->GetSocket());
