--- a/server/metin2/Source/Server/db/src/GuildManager.cpp
+++ b/server/metin2/Source/Server/db/src/GuildManager.cpp
@@ -5,6 +5,7 @@
 #include "QID.h"
 #include "Config.h"
 #include <math.h>
+#include <climits>
 
 extern std::string g_stLocale;
 
@@ -721,10 +722,16 @@
 
 	TGuild& r = it->second;
 
-	r.ladder_point += change;
+	r.ladder_point = [&]()
+	{
+		const int64_t llNewPoint = static_cast<int64_t>(r.ladder_point) + static_cast<int64_t>(change);
+		if (llNewPoint < 0)
+			return 0;
+		if (llNewPoint > INT_MAX)
+			return INT_MAX;
+		return static_cast<int>(llNewPoint);
+	}();
 
-	if (r.ladder_point < 0)
-		r.ladder_point = 0;
 
 	char buf[1024];
 	snprintf(buf, sizeof(buf), "UPDATE guild%s SET `ladder_point` = %d WHERE `id` = %u", GetTablePostfix(), r.ladder_point, GID);
