--- a/home/metin2/Source/Server/game/src/messenger_manager.cpp
+++ b/home/metin2/Source/Server/game/src/messenger_manager.cpp
@@ -232,8 +232,10 @@
 	strlcpy(sCompanionData.szCountry, pkCharCompanion ? pkCharCompanion->GetCountry() : "", sizeof(sCompanionData.szCountry));
 #endif
 
-	m_map_strRelation[c_strAccount].emplace_back(c_strCompanion, sCompanionData);
-	m_map_strInverseRelation[c_strCompanion].emplace_back(c_strAccount, sCompanionData);
+	m_map_strRelation[c_strAccount].reserve(64);
+			m_map_strRelation[c_strAccount].emplace_back(c_strCompanion, sCompanionData);
+	m_map_strInverseRelation[c_strCompanion].reserve(64);
+			m_map_strInverseRelation[c_strCompanion].emplace_back(c_strAccount, sCompanionData);
 #endif
 
 	if (pkCharCompanion)
