diff -uNr "/mnt/data/nexus_orig/Nexus OS/server/metin2/Source/Server/game/src/char_item.cpp" /mnt/data/tmp_attr67_small/game/src/char_item.cpp
--- "/mnt/data/nexus_orig/Nexus OS/server/metin2/Source/Server/game/src/char_item.cpp"	2026-02-16 17:22:47.208543568 +0000
+++ /mnt/data/tmp_attr67_small/game/src/char_item.cpp	2026-02-16 18:17:40.122448467 +0000
@@ -11729,8 +11729,13 @@
 	if (pkRegistItem->GetType() != ITEM_ARMOR && pkRegistItem->GetType() != ITEM_WEAPON)
 		return false;
 
+	pkRegistItem->Lock(true);
+
 	if (pkRegistItem->GetAttributeCount() < ITEM_MANAGER::MAX_NORM_ATTR_NUM)
+	{
+		pkRegistItem->Lock(false);
 		return false;
+	}
 
 	DWORD dwItemMaterialVnum = pkRegistItem->Get67AttrMaterial();
 	if (CountSpecifyItem(dwItemMaterialVnum) < kAttr67AddData.byMaterialCount)
@@ -11765,7 +11770,10 @@
 	if (pSupportItem)
 	{
 		if (CountSpecifyItem(pSupportItem->GetVnum()) < kAttr67AddData.bySupportItemCount)
+		{
+			pkRegistItem->Lock(false);
 			return false;
+		}
 
 		RemoveSpecifyItem(pSupportItem->GetVnum(), kAttr67AddData.bySupportItemCount);
 	}
@@ -11777,6 +11785,7 @@
 		if (!pkAttr67Add)
 		{
 			// TODO: Make a backup of the item in case something goes bad.
+			pkRegistItem->Lock(false);
 			sys_err("CHARACTER::Attr67Add: failed to get regist item from ATTR67_ADD (window).");
 			return false;
 		}
diff -uNr "/mnt/data/nexus_orig/Nexus OS/server/metin2/Source/Server/game/src/questlua_attr67add.cpp" /mnt/data/tmp_attr67_small/game/src/questlua_attr67add.cpp
--- "/mnt/data/nexus_orig/Nexus OS/server/metin2/Source/Server/game/src/questlua_attr67add.cpp"	2026-02-16 17:22:47.612905967 +0000
+++ /mnt/data/tmp_attr67_small/game/src/questlua_attr67add.cpp	2026-02-16 18:17:40.123154590 +0000
@@ -48,6 +48,7 @@
 				{
 					pkItem->RemoveFromCharacter();
 					pkChar->SetItem(TItemPos(INVENTORY, iEmptyPos), pkItem);
+					pkItem->Lock(false);
 
 					// After collecting, reset all quest flags.
 					pkChar->SetQuestFlag("add_attr67.success", 0);
