--- a/server/metin2/Source/Server/libsql/AsyncSQL.cpp
+++ b/server/metin2/Source/Server/libsql/AsyncSQL.cpp
@@ -437,12 +437,24 @@
 
 DWORD CAsyncSQL::CountQuery()
 {
-	return m_queue_query.size();
+	if (!m_mtxQuery)
+		return m_queue_query.size();
+
+	MUTEX_LOCK(m_mtxQuery);
+	DWORD dwSize = m_queue_query.size();
+	MUTEX_UNLOCK(m_mtxQuery);
+	return dwSize;
 }
 
 DWORD CAsyncSQL::CountResult()
 {
-	return m_queue_result.size();
+	if (!m_mtxResult)
+		return m_queue_result.size();
+
+	MUTEX_LOCK(m_mtxResult);
+	DWORD dwSize = m_queue_result.size();
+	MUTEX_UNLOCK(m_mtxResult);
+	return dwSize;
 }
 
 void __timediff(struct timeval* a, struct timeval* b, struct timeval* rslt)
